### PR TITLE
[SW-2235] Setup infra for running tests on Kubernetes

### PR DESCRIPTION
**Pull request by jakubhava**
I have decided to implement this in a few smaller PRs so it is easier to review.

The goal of this PR was to:
 -> Add Job to our Jenkins to test Kubernetes
-> Fix docker images
-> Create Jenkins pipeline which starts Kubernetes on AWS, builds SW and uploads images to created docker registry and destroy the Kubernetes cluster after it's done.
-> Ensure Kubernetes is not part of doc for unsupported versions

The next PR should be focused on writing the tests themselfs

As a verification -> the job is here https://sparkling-jenkins.h2o.ai/blue/organizations/jenkins/TESTS_KUBERNETES/detail/jh%2FSW-2235/45/pipeline/174/ It failed in last step ( tearing down EKS), but that is fixed by https://github.com/h2oai/sparkling-water/pull/2119/commits/a0f54df0fae5c2ac1516666eb29039506c310ef6. The job run before this commit

### CODE REVIEW

**Comment by jakubhava** [comment]
Triggered new job based on the latest state https://sparkling-jenkins.h2o.ai/blue/organizations/jenkins/TESTS_KUBERNETES/detail/jh%2FSW-2235/46/pipeline

File: ci/Jenkinsfile-kubernetes [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+properties(
+        [
+                pipelineTriggers([cron('H 16 * * *')]),
+                buildDiscarder(logRotator(numToKeepStr: '10'))
+        ]
+)
+
+def getKubernetesSparkVersions() {
+    def versionLine = readFile("gradle.properties").split("\n").find() { line -> line.startsWith('supportedSparkVersions') }
```
Comment (mn-mikke) [inline ci/Jenkinsfile-kubernetes:12]: Is it possible to read the `gradle.properties` file once? E.g convert to a map first and then use it?

File: kubernetes/build.gradle [hunk 2]
```diff
@@ -21,8 +22,19 @@ task createScalaDockerfile(type: Dockerfile) {
...
   destFile = outputFileR
   arg "spark_version"
-  from 'spark-r:$spark_version'
-  runCommand "apk add libc-dev linux-headers g++ libxml2-dev"
+  if (VersionNumber.parse(sparkVersion) >= VersionNumber.parse("2.4.0")) {
+    from 'spark-r:$spark_version'
+  } else {
+    from 'spark:$spark_version'
+  }
+  if (VersionNumber.parse(sparkVersion) >= VersionNumber.parse("2.4.4")) {
```
Comment (mn-mikke) [inline kubernetes/build.gradle:30]: Shouldn't be there `2.4.5` on this line?

File: kubernetes/build.gradle [hunk 3]
```diff
@@ -41,14 +53,25 @@ task createRDockerfile(type: Dockerfile) {
...
+    from 'spark:$spark_version'
+  }
+  if (VersionNumber.parse(sparkVersion) < VersionNumber.parse("2.4.0")) {
+    runCommand "apk add --update py-pip"
+  }
   copyFile("h2o_pysparkling_${sparkMajorVersion}-${version}.zip", "/opt/spark/pyspark/python/lib/h2o_pysparkling_${sparkMajorVersion}-${version}.zip")
-  runCommand """\\
-                pip install /opt/spark/pyspark/python/lib/h2o_pysparkling_${sparkMajorVersion}-${version}.zip && \\
+  runCommand "pip install /opt/spark/pyspark/python/lib/h2o_pysparkling_${sparkMajorVersion}-${version}.zip"
+  if (VersionNumber.parse(sparkVersion) >= VersionNumber.parse("2.4.4")) {
```
Comment (mn-mikke) [inline kubernetes/build.gradle:66]: Shouldn't be there `2.4.5` on this line?

File: kubernetes/build.gradle (hunk 2 above)
Comment (jakubhava) [reply kubernetes/build.gradle:30]: Yes, good catch!

File: kubernetes/build.gradle (hunk 3 above)
Comment (jakubhava) [reply kubernetes/build.gradle:66]: Yes, good catch!

File: ci/Jenkinsfile-kubernetes (hunk 1 above)
Comment (jakubhava) [reply ci/Jenkinsfile-kubernetes]: Good idea, done. I re-triggered the job to verify it works after this change

**Pull request merged**